paymentsheet/src/main/java/com/stripe/android/customersheet/data/CustomerAdapterDataSource.kt

```diff
@@ -0,0 +1,23 @@
+package com.stripe.android.customersheet.data
+
+import com.stripe.android.customersheet.CustomerAdapter
+import com.stripe.android.customersheet.ExperimentalCustomerSheetApi
+import com.stripe.android.customersheet.map
+import com.stripe.android.model.PaymentMethod
+import com.stripe.android.paymentsheet.model.SavedSelection
+import javax.inject.Inject
+
+@OptIn(ExperimentalCustomerSheetApi::class)
+internal class CustomerAdapterDataSource @Inject constructor(
+    private val customerAdapter: CustomerAdapter,
+) : CustomerSheetSavedSelectionDataSource, CustomerSheetPaymentMethodDataSource {
```

Comment:
I'm not sure there's much value in having separate data sources if the implementation is going to be in one class anyways. Do you think that this will look the same for the customer session adapter or will the implementation be more complex?

Reply:
Yeah it will end up being one data source. Especially since we will be sharing an ephemeral key between most of the operations. Should we go back to having one data source since the implementation is one data source anyways?

Reply:
talked offline and it makes sense to still keep the separate interfaces as it'll be useful for customer session

Reply:
Talked about this in-person, the `CustomerAdapterDataSource` will remain one data source but we will split it out the `CustomerSession` implementation.